R/learner_lightgbm_regr_lightgbm.R

```diff
@@ -236,11 +330,109 @@ LearnerRegrLightGBM = R6Class("LearnerRegrLightGBM",
 
       list(response = pred)
     },
+
     .hotstart = function(task) {
       pars = self$param_set$get_values(tags = "train")
       pars_train = self$state$param_vals
+
+      if (!is.null(pars_train$early_stopping_rounds)) {
+        stop("The parameter `early_stopping_rounds` is set. Early stopping and hotstarting are incompatible.")
+      }
+
+      # calculate additional boosting iterations
       pars_train$num_iterations = pars$num_iterations - self$state$param_vals$num_iterations
-      train_lightgbm(self, task, "classif", pars_train, self$model)
+
+      # convert data
+      x_train = data.matrix(task$data(rows = task$row_roles$use, cols = task$feature_names))
+      y_train = task$data(rows = task$row_roles$use, cols = task$target_names)[[1L]]
+
+      # create data set
+      categorical_feature = if (any(task$feature_types$type %in% c("factor", "logical"))) {
+        task$feature_types$id[task$feature_types$type %in% c("factor", "logical")]
+      }
+
+      dtrain = lightgbm::lgb.Dataset(
+        data = x_train,
+        label = y_train,
+        free_raw_data = FALSE,
+        categorical_feature = categorical_feature
+      )
+
+      if ("weights" %in% task$properties) {
+        dtrain$set_field("weight", task$weights[get("row_id") %in% task$row_roles$use, "weight"][[1L]])
+      }
+
+      ii = names(pars_train) %in% formalArgs(lightgbm::lgb.train)
+      args = pars_train[ii]
+      params = pars_train[!ii]
+
```

Comment:
here the code for validation is missing. I also don't understand the advantage of not using `train_lightgbm` anymore, we are just duplicating code?

Reply:
Done